File: Capítulo_7/ECMAScript 2015 – a classe Set/Operações Set ES2015/Simulando a operação de diferença/difference.js
// Usaremos os dois conjuntos a seguir em nossos exemplos:
const setA = new Set();
setA.add(1);
setA.add(2);
setA.add(3);
const setB = new Set();
setB.add(2);
setB.add(3);
setB.add(4)

const difference = (set1, set2) => {
    const differenceSet = new Set();
    set1.forEach(value => {
        if (!set2.has(value)) {
            differenceSet.add(value);
        }
    });
    return differenceSet;
}
console.log(difference(setA, setB)); // Set(1) { 1 }

File: Capítulo_7/ECMAScript 2015 – a classe Set/Operações Set ES2015/Simulando a operação de diferença/Código em Python/difference.py
# Usaremos os dois conjuntos a seguir em nossos exemplos:
setA = set()
setA.add(1)
setA.add(2)
setA.add(3)
setB = set()
setB.add(2)
setB.add(3)
setB.add(4)

def difference(set1, set2):
    difference_set = set()
    for value in set1:
        if value not in set2:
            difference_set.add(value)
    return difference_set

difference_result = difference(setA, setB)

print(difference_result) # { 1 }
